packages/ts/react-crud/src/autoform.tsx

```diff
@@ -313,29 +234,37 @@ export function ExperimentalAutoForm<M extends AbstractModel>({
 
   function createAutoFormField(propertyInfo: PropertyInfo): JSX.Element {
     const fieldOptionsForProperty = fieldOptions?.[propertyInfo.name];
+    const colspanValue = fieldOptionsForProperty?.colspan;
+    const colspan = colspanValue ? { colspan: colspanValue } : {};
+
     return (
       <AutoFormField
         key={propertyInfo.name}
         propertyInfo={propertyInfo}
         form={form}
         disabled={disabled}
         options={fieldOptionsForProperty}
+        {...colspan}
       />
     );
   }
 
-  let layout: JSX.Element;
-  if (CustomLayoutRenderer === undefined) {
-    const fields = getProperties(model).filter(includeProperty).map(createAutoFormField);
-    layout = <FormLayout>{fields}</FormLayout>;
-  } else {
-    const fields = getProperties(model).map(createAutoFormField);
-    if (typeof CustomLayoutRenderer === 'function') {
-      layout = <CustomLayoutRenderer form={form}>{fields}</CustomLayoutRenderer>;
-    } else {
-      layout = <CustomFormLayout customFormLayout={CustomLayoutRenderer}>{fields}</CustomFormLayout>;
-    }
-  }
+  const defaultProperties = getProperties(model);
+
+  const visibleProperties = visibleFields
+    ? visibleFields.reduce<PropertyInfo[]>((foundProperties, propertyName) => {
+        const maybeProperty = defaultProperties.find((p) => p.name === propertyName);
+        return maybeProperty ? [...foundProperties, maybeProperty] : foundProperties;
+      }, [])
+    : defaultProperties.filter(includeProperty);
```

Comment:
It looks like this would allow you to specify properties that `AutoFormField` doesn't know how to render. In comparison, in auto grid we first filter the default properties on whether they may be included, and only from that derive the visible columns.

OTOH you can provide a custom renderer with `fieldOptions`, so still a valid use case I guess?

Reply:
Yes, `AutoForm` will try to include all fields listed in `visibleFields`. Anyway, if there's no custom renderer, the `AutoFormField` will be undefined for unhandled properties, so the filtering you mentioned still happens.

Reply:
Right, should be fine then.